ibis/backends/base/file/__init__.py

```diff
@@ -134,7 +136,9 @@ def connect(self, path):
     def version(self) -> str:
         return pd.__version__
 
-    def list_tables(self, path: Path = None, like: str = None):
+    def list_tables(
+        self, path: Path = None, like: str = None, database: str = None
```

Comment:
We should eventually require `Optional` here or use the new annotations syntax of `T | None`, but that can be done in a separate PR.

Reply:
Afaik mypy is smart enough to consider anything having a default value of `None` as optional, I don't think it's needed.

Reply:
Unfortunately, mypy's ability to infer types is unrelated to the requirement for `Optional` or an equivalent syntax.

`Path` is simply not equivalent to `Optional[Path]`. If an API accepts `Path`, it should not accept `None`, because `Path` doesn't sufficiently model a option type.